plugins/backend-trace-forwarder/src/Cardano/BM/Backend/TraceForwarder.lhs

```diff
@@ -95,12 +105,13 @@ newtype TraceForwarder a = TraceForwarder
 type TraceForwarderMVar a = MVar (TraceForwarderInternal a)
 
 data TraceForwarderInternal a = TraceForwarderInternal
-    { tfQueue      :: TBQ.TBQueue (LogObject a)
-    , tfHandle     :: Maybe Handle
-    , tfRemoteAddr :: RemoteAddr
-    , tfFilter     :: Severity
-    , tfDispatcher :: Maybe (Async.Async ())
+    { tfQueue            :: TBQ.TBQueue (LogObject a)
+    , tfHandle           :: Maybe Handle
+    , tfRemoteAddr       :: RemoteAddr
+    , tfFilter           :: Severity
+    , tfDispatcher       :: Maybe (Async.Async ())
     , tfQueueFullCounter :: IORef Int
+    , tfDirectLogObjects :: IO [LogObject a]
```

Comment:
I would say that the `Direct` part in the name is confusing -- it doesn't seem to explain much.

What is the essence of those `LogObjects`?

Here's what comes to mind:

- rarely changed
- rarely logged
- important parameters
- it's all accumulated state
- we will log them once

There's a notion that could work well here, I think -- `Digest`.  And since we're dealing with state -- `StateDigest`.

Also, I don't think we need to duplicate `LogObjects` in the name -- it's already in the type.

So:

```
  tfGetStateDigest :: IO [LogObject a]
```

What do you think?

Reply:
Agree. Done.